lib/src/view/broadcast/broadcast_screen.dart

```diff
@@ -84,104 +85,86 @@ class _Body extends ConsumerWidget {
   }
 }
 
-const _kGridPadding = 10.0;
-
 class BroadcastPreview extends StatelessWidget {
-  final Iterable<BroadcastGameSnapshot>? games;
+  final List<BroadcastGameSnapshot>? games;
 
   const BroadcastPreview({super.key, this.games});
 
   @override
   Widget build(BuildContext context) {
-    const numberLoadingBoardRows = 6;
+    const numberLoadingBoards = 12;
+
+    const textHeight = 19.0;
+
+    const headerAndFooterPadding = EdgeInsets.symmetric(vertical: 5);
+
+    const boardSpacing = 10.0;
+
+    final numberOfBoardsByRow = isTabletOrLarger(context) ? 4 : 2;
+
+    final screenWidth = MediaQuery.of(context).size.width;
+
+    final boardWidth = (screenWidth -
+            Styles.bodyPadding.horizontal -
+            (numberOfBoardsByRow - 1) * boardSpacing) /
+        numberOfBoardsByRow;
+
     final fakeHeaderAndFooter = Padding(
-      padding: const EdgeInsets.symmetric(vertical: 5),
+      padding: headerAndFooterPadding,
       child: Container(
-        height: 24,
+        height: textHeight,
         decoration: BoxDecoration(
           color: Colors.black,
           borderRadius: BorderRadius.circular(5),
         ),
       ),
     );
 
-    final crossAxisCount =
-        MediaQuery.sizeOf(context).width > FormFactor.tablet ? 4 : 2;
-    final columnsGap = _kGridPadding * crossAxisCount + _kGridPadding;
-    final boardWidth =
-        (MediaQuery.sizeOf(context).width - columnsGap) / crossAxisCount;
-
-    final List<Iterable<BroadcastGameSnapshot>> list = [];
-    if (games != null) {
-      list.addAll(games!.slices(crossAxisCount));
-    }
-
     return SafeArea(
-      child: ListView.builder(
-        itemCount: list.isEmpty ? numberLoadingBoardRows : list.length,
-        itemBuilder: (context, index) {
-          final itemPadding = EdgeInsets.only(
-            left: _kGridPadding,
-            top: _kGridPadding / 2,
-            bottom: index == list.length - 1 ? _kGridPadding : 0,
-          );
+      child: Padding(
+        padding: Styles.bodyPadding,
+        child: GridView.builder(
```

Comment:
The number of elements is not that large, I wonder if the `GridView.count` would be more appropriate? The code would be a little easier to read I think.

Reply:
I don't know how it would affect performances. According to the `GridView.builder` documentation `the builder is called only for those children that are actually visible`. From what I understand the `GridView.count` will create all widgets at once but it will only render them when they are visible on the screen whereas the `GridView.builder` will create only the widgets that are needed.

Does the number of elements that is not that large applies to the puzzles too ?

Reply:
No the puzzle history screen is paginated and thus potentially very large.

But the games number in a broadcast round are usually quite limited, like usually less than 50 and I doubt there are such big tournaments that this number can be more than like 100 games.

Reply:
Ok I'll use the `GridView.count`

Reply:
Actually I don't think it will make the code more readable because I don't have access anymore to `mainAxisExtent` so I would need to use `childAspectRatio` and also I would need to use `List.generate` to generate grid widgets.